### DISCUSSION TITLE
Modbus write question

### DISCUSSION THREAD

**Discussion by einarbjorkman** [artisan-roaster-scope/artisan#964] [Q&A]
Hello,
I am making use of the Modbus write command when I press CHARGE.
I write the value 1 to a register and then have a IF loop in my program that runs if the register == 1.
At the end of the loop I set the register = 0.
However it seems that Artisan continuously writes 1 to the register after pressing CHARGE instead of only one time. 
For my application this becomes quite inconvenient. 
Is there any way to modify this behavior so that Artisan only writes to the register one time when I press CHARGE?

/ Einar

**Comment by MAKOMO** [comment]
Button actions are only issued on releasing the button. Maybe you are using a different configuration than I have in mind. I could check if you would have shared your Artisan configuration (save via menu Help >> Save Settings, rename to end with .txt and attach it here via drag-and-drop to the GitHub comment section).

**Comment by einarbjorkman** [comment]
Ok, here they come:
[artisan-settings.txt](https://github.com/artisan-roaster-scope/artisan/files/9640506/artisan-settings.txt)
Thanks for your reply!

**Reply by MAKOMO** [reply]
Looks all good. I cannot reproduce your observation that Artisan is continuously writing 1 to the indicated registered. This happens only one time on releasing the CHARGE button.

**Reply by einarbjorkman** [reply]
Ok, anyhow I have been able to solve it with a "register counter", so that every time the program passes through the main loop and sees that the register == 1, the regCounter increases. I have then conditioned my IF loops based on this which provides a workaround solution.

But now that you have my settings, I am actually having another urgent problem:
When I save a custom made curve from designer and open it, the curve looks completley different. 
Some semblance still exists but some points are missing, others have been changed by 10s of degrees or multiple minutes, etc...
Any ideas?
Fast help would be very very appreciated!

/ Einar

**Reply by einarbjorkman** [reply]
Actually I think we got it by:
Saving points in designer
Switching out of designer mode
Saving as alog file

Cheers / Einar

**Reply by MAKOMO** [reply]
What you write is all confusing to me. What I know, as I implemented and tested it, the CHARGE action is only issued once at the moment the CHARGE button is released.
